README.md

```diff
@@ -1,44 +1,39 @@
 [![Documentation Status](https://readthedocs.org/projects/goblint/badge/?version=latest)](https://goblint.readthedocs.io/en/latest/?badge=latest)
+[![Docker Build Status](https://img.shields.io/docker/cloud/build/voglerr/goblint)](https://hub.docker.com/r/voglerr/goblint)
 
 Documentation can be browsed on [Read the Docs](https://goblint.readthedocs.io/en/latest/) or [GitHub](./docs/).
 
-## Setup
-### Linux / MacOS
-**For an up-to-date-version, clone this repository, use `make setup` to install OCaml and all dependencies, and `make` to build.**
-
-Alternatively: Install [opam](http://opam.ocaml.org/doc/Install.html), and then do `opam install goblint`. Warning: The OPAM package is updated infrequently.
+## Installing
+Both for using an up-to-date version of Goblint or developing it, the best way is to install from source by cloning this repository.
 
-Run goblint: `./goblint tests/regression/04-mutex/01-simple_rc.c`.
+_The [goblint package on opam](https://opam.ocaml.org/packages/goblint/) is very outdated and should currently not be used._
 
-If something goes wrong, take a look at [travis-ci.sh](scripts/travis-ci.sh) for an example Ubuntu/macOS setup.
+### Linux
+1. Install [opam](https://opam.ocaml.org/doc/Install.html).
+2. Run `make setup` to install OCaml and dependencies via opam.
+3. Run `make` to build Goblint itself.
 
-### macOS
-Goblint relies on GNU `cpp` to preprocess source files - the default clang `cpp` on macOS will not work.
-You can install it with `brew install gcc` (first do `xcode-select --install` if you don't want to build from source). You can check src/config.ml to see what command is used to call `cpp`.
+### MacOS
+1. Install GCC with `brew install gcc` (first run `xcode-select --install` if you don't want to build it from source). Goblint requires GCC while macOS's default `cpp` is Clang, which will not work.
+2. Continue using Linux instructions.
 
 ### Windows
-For Windows, we recommend using [WSL](https://docs.microsoft.com/de-de/windows/wsl/install-win10).
-
-### Docker / Virtual machine
-You can run Goblint in a [Docker container](https://hub.docker.com/r/voglerr/goblint/) using: `docker run -it voglerr/goblint ./goblint --help`.
-
-A virtual machine containing Goblint can be set up using [Vagrant](http://www.vagrantup.com/): `vagrant up && vagrant ssh`.
+1. Install [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+2. Continue using Linux instructions in WSL.
 
-### Web frontend
-The analysis results are printed to stdout by default.
-Adding `--html` saves the results as XML, which is then transformed to be viewable in a web browser.
-Use `make jar` to build the needed Java program.
+### Other
+* **[Vagrant](https://www.vagrantup.com/).** Run `vagrant up && vagrant ssh`.
+* **[Docker](https://hub.docker.com/r/voglerr/goblint)** (outdated). Run `docker run -it voglerr/goblint bash`.
```

Comment:
The last build on Docker Hub is a month old by now, probably because the free Docker Cloud tier was shut down, so no new ones are automatically built and published.

Maybe we should just have here the instructions for building the Docker image from the cloned repository? I don't know the right way to do that.

Reply:
@vogler ?

Reply:
> Maybe we should just have here the instructions for building the Docker image from the cloned repository? I don't know the right way to do that.

`make docker`?

Reply:
> The last build on Docker Hub is a month old by now, probably because the free Docker Cloud tier was shut down, so no new ones are automatically built and published.

I applied for their Open Source Program but they are very slow to respond. Got this yesterday despite having set the license in their form (I see no license field on the website):

 "Thanks again for applying to Docker's Open Source Program. We noticed that you didn't specify the open source license under which your image namespace is under. As a reminder, to qualify for the program, images need to be under a license approved by the OSI."

Reply:
I added `make docker` as a non-outdated variant and left Docker Hub as an outdated one for now. This should be clear enough that it's possible to get a Docker build with the right environment, but not to suggest the image from Docker Hub first.